woocommerce/woocommerce
# Move `ReportError` to `@woocommerce/components` as `AnalyticsError`

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Move ReportError to `@woocommerce/components` as `<AnalyticsError>`
This is the first PR that takes the easiest, lowest-hanging fruit to pave the path for other `<Report*>` components. This one has no dependencies, so should be easier to review so we can focus on the folder structure, etc.

Address part of https://github.com/woocommerce/woocommerce/issues/36052

The goal is to publish those components to be reused by other extensions migrating their legacy reports to the new Analytics.

I decided to rename components to `<Analytics*>` as `@woocommerce/components` already have a `<ReportFilters>` which are actually used inside `plugins/woocommerce-admin/client/analytics/components/report-filters/index.js`.
So the current `@woocommerce/components ReportFilters` are a bit more general-purpose, while `woocommerce-admin/.../analytics... ReportFilters` are specifically for analytics.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout and build this branch
2. Visit any analytics page with an error. You can simulate that by providing malformed query param, like `/wp-admin/admin.php?page=wc-admin&path=%2Fanalytics%2Fcategories&chart=net_revenue&orderby=net_revenue&order=foo`
3. Build and [check storybook](http://localhost:6007/?path=/story/woocommerce-admin-components-analytics-analyticserror--basic)
```
cd tools/storybook
npm run storybook
```

![image](https://github.com/user-attachments/assets/18b1eb96-c236-4ff2-974a-369066fcab2f)





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Publish `<ReportError>` to `Woocommerce/components`

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
